apps/www/components/react-hook-form/form.tsx

```diff
@@ -44,24 +42,23 @@ const FormField = <
 const useFormField = () => {
   const fieldContext = React.useContext(FormFieldContext)
   const itemContext = React.useContext(FormItemContext)
-  const controller = useController(fieldContext)
+  const { getFieldState, formState } = useFormContext()
+
+  const fieldState = getFieldState(fieldContext.name, formState)
```

Comment:
Ha! Did not know you could do that. I mean I assumed `useController` was easier within a `Controller`. Thanks

Reply:
The `getFieldState` method differs slightly from hooks + `formState` being a Proxy: this is a little more convoluted but I'd argue it's fine being in a library code 😊

But, yea, `useController` is a way to actually register your field with the form instead of hooking into it in different places in your component tree. Sadly, I don't have the time to explore the alternative APIs to make it clearer, which I really want to do some time!